### ISSUE TITLE
Decision Tree

### ISSUE THREAD

**Issue by ysh329** [yuenshome/yuenshome.github.io#47]

**Comment by ysh329** [comment]
gbdt原理(非常重要)_图文_百度文库
https://wenku.baidu.com/view/f3da60b4951ea76e58fafab069dc5022aaea463e.html

**Comment by ysh329** [comment]
GBDT算法原理深入解析_图文_百度文库
https://wenku.baidu.com/view/4d008fa7b8f3f90f76c66137ee06eff9aef84923.html?rec_flag=default&sxts=1545006778033

**Comment by ysh329** [comment]
GBDT 算法：原理篇 - 云+社区 - 腾讯云
https://cloud.tencent.com/developer/article/1005611

**Comment by ysh329** [comment]
## 内容

![image](https://user-images.githubusercontent.com/7320657/50066922-5d945f00-01f9-11e9-98dc-92f3ca323f63.png)

## 泰勒公式

![image](https://user-images.githubusercontent.com/7320657/50066944-7ac92d80-01f9-11e9-9fb4-d61ea4a87364.png)

## 梯度下降法

![image](https://user-images.githubusercontent.com/7320657/50066998-bb28ab80-01f9-11e9-8ee2-e67f53f522e4.png)

## 牛顿法

![image](https://user-images.githubusercontent.com/7320657/50067026-df848800-01f9-11e9-85f9-6edcc67a1bf4.png)


## 从参数空间到函数空间

![image](https://user-images.githubusercontent.com/7320657/50067049-f1662b00-01f9-11e9-9230-fc3fe656b0ce.png)

## 从gradient descent到gradient boosting

![image](https://user-images.githubusercontent.com/7320657/50067076-1490da80-01fa-11e9-8008-1e6998934718.png)


## 从Newton'method到Newton's boosting

![image](https://user-images.githubusercontent.com/7320657/50067097-312d1280-01fa-11e9-9d12-1e65809e7b41.png)

## 小结

![image](https://user-images.githubusercontent.com/7320657/50067109-4609a600-01fa-11e9-8aba-348be3c40222.png)

**Comment by ysh329** [comment]
# Gradient Boosting Tree算法原理

![image](https://user-images.githubusercontent.com/7320657/50067134-81a47000-01fa-11e9-8030-f077d61672f5.png)

----

![image](https://user-images.githubusercontent.com/7320657/50067163-aac50080-01fa-11e9-992c-4d74bc44bf5c.png)

**Comment by ysh329** [comment]
## Newton Boosting Tree算法原理：详解XGBoost

![image](https://user-images.githubusercontent.com/7320657/50067303-5706e700-01fb-11e9-8c57-50ffddbdc5d3.png)

----

### 模型函数形式

![image](https://user-images.githubusercontent.com/7320657/50067318-6e45d480-01fb-11e9-8f4d-ed4b83a12e49.png)

----

![image](https://user-images.githubusercontent.com/7320657/50067326-7b62c380-01fb-11e9-8ae4-8f054be4ce5d.png)

### 目标函数

![image](https://user-images.githubusercontent.com/7320657/50067343-9f260980-01fb-11e9-9c59-06bff09c1b87.png)

**Comment by ysh329** [comment]
## 回归树的学习策略

![image](https://user-images.githubusercontent.com/7320657/50067623-ae598700-01fc-11e9-8ec9-371e1eba4ce5.png)


### XGBoost的打分函数

![image](https://user-images.githubusercontent.com/7320657/50067636-bd403980-01fc-11e9-96bd-2447e57220f9.png)


### 树节点分裂方法（Split Finding）

![image](https://user-images.githubusercontent.com/7320657/50067651-d77a1780-01fc-11e9-981e-3cf3feacc682.png)

----

![image](https://user-images.githubusercontent.com/7320657/50067663-e660ca00-01fc-11e9-833c-357111127df3.png)

----

![image](https://user-images.githubusercontent.com/7320657/50067673-f7114000-01fc-11e9-957d-c9f722cbbd99.png)

---

![image](https://user-images.githubusercontent.com/7320657/50067694-0d1f0080-01fd-11e9-83d4-e9b9f9152e7d.png)

**Comment by ysh329** [comment]
## 缺失值的处理

![image](https://user-images.githubusercontent.com/7320657/50067720-3c357200-01fd-11e9-9d80-dae7fe651e0a.png)

**Comment by ysh329** [comment]
## XGBoost其它特性

![image](https://user-images.githubusercontent.com/7320657/50067743-5ff8b800-01fd-11e9-8fae-162abd98a836.png)


## 更高效的LightGBM

![image](https://user-images.githubusercontent.com/7320657/50067763-83236780-01fd-11e9-8bbd-25d8d60c782e.png)

## LightGBM相比XGBoost的改进

![image](https://user-images.githubusercontent.com/7320657/50067820-c251b880-01fd-11e9-91ad-bc378b4e966b.png)

## 参考

![image](https://user-images.githubusercontent.com/7320657/50067843-d4335b80-01fd-11e9-9dbb-e8a286942196.png)
